common/network-common/pom.xml

```diff
@@ -61,6 +61,26 @@
       <artifactId>netty-transport-native-kqueue</artifactId>
       <classifier>osx-x86_64</classifier>
     </dependency>
+    <dependency>
```

Comment:
Does the `network-yarn` module have a requirement to use these native codes?

Reply:
I don't believe it needs a direct dependency here if you look at the code changes to `resource-managers/yarn` in https://github.com/apache/spark/pull/42685 - we do add tests to the yarn code that do pass.

We do change `ShuffleTransportContext` which is in `network-common`, and that's what is used by yarn here. `network-yarn` depends on `network-shuffle` which depends on `network-common` so I think that's how the dependency gets pulled in.

Reply:
These need to be test scope right?

Reply:
@srowen the `bouncycastle` and `network-common-helper` dependencies are only needed in the test scope, but the `netty-tcnative-boringssl-static` dependencies need to be available in the production build. We need this dependency in order to provide SSL support.